python/ray/ml/storage.py

```diff
@@ -0,0 +1,183 @@
+import abc
+import subprocess
+from typing import Dict
+
+from ray import logger
+from ray.util.annotations import DeveloperAPI
+
+S3_PREFIX = "s3://"
+GS_PREFIX = "gs://"
+HDFS_PREFIX = "hdfs://"
+ALLOWED_REMOTE_PREFIXES = (S3_PREFIX, GS_PREFIX, HDFS_PREFIX)
+
+
+class Storage(abc.ABC):
```

Comment:
Is this planned to replace sync_client sync template stuff?

Reply:
I think eventually that would be good, but this will have some problems, as those have to be propagated to remote nodes. So in the mid run, yes, but not in the short run.